### PR TITLE
Kubermatic controller manager controller: Set seed reconcile success condition

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Makes all the controllers in the kubermatic controller manager set a `reconcileSuccess` condition, which we need for #3417 

I think I'll create follow-ups for:
* Adding a variadic arg `predicates ..func(*cluster)(*reconcile.Result, error)` to the `ClusterReconcileWrapper`  which we can use for controllers that have extra conditions, e.G. openshift/kubernetes or the `ClusterAvailableForReconciling`
* Move the whole `SetSeedResourcesUpToDateCondition` into its own controller and remove the `successfullyReconciled` bit from it, as this is a condition that spans multiple controller and can currently cause racing if one controller fails to deploy some parts

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```
/assign @xrstf

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -107,6 +107,21 @@ const (
...
 	ClusterConditionSeedResourcesUpToDate ClusterConditionType = "SeedResourcesUpToDate"
 
+	// ClusterConditionClusterControllerReconcilingSuccess indicates whether the cluster
+	// controller successfully finished reconciling this cluster.
+	ClusterConditionClusterControllerReconcilingSuccess ClusterConditionType = "ClusterControllerReconciledSuccessfully"
+
+	// ClusterConditionAddonControllerReconcilingSuccess indicates whether the addon controller
+	// succeeded in reconciling the cluster.
+	ClusterConditionAddonControllerReconcilingSuccess          ClusterConditionType = "AddonControllerReconciledSuccessfully"
+	ClusterConditionAddonInstallerControllerReconcilingSuccess ClusterConditionType = "AddonInstallerControllerReconciledSuccessfully"
```
Comment (xrstf) [inline api/pkg/crd/kubermatic/v1/cluster.go:112]: And at this line you just gave up adding comments? :-D I don't mind missing comments on these self-explanatory constants, just wondering about the inconsistent use.

Comment (alvaroaleman) [reply api/pkg/crd/kubermatic/v1/cluster.go:112]: Yeah, pretty much. Do you want all with or all without comment?

File: api/pkg/controller/addon/addon_controller.go [hunk 2]
```diff
@@ -149,42 +150,44 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
...
-		reconcilingError := err
-		//Get the cluster so we can report an event to it
-		cluster := &kubermaticv1.Cluster{}
-		if err := r.Get(ctx, types.NamespacedName{Name: addon.Spec.Cluster.Name}, cluster); err != nil {
-			log.Errorw("failed to get cluster for reporting error onto it", zap.Error(err))
-		} else {
-			r.recorder.Eventf(cluster, corev1.EventTypeWarning, "ReconcilingError",
-				"failed to reconcile Addon %q: %v", addon.Name, reconcilingError)
-		}
+		r.recorder.Event(addon, corev1.EventTypeWarning, "ReconcilingError", err.Error())
```
Comment (xrstf) [inline api/pkg/controller/addon/addon_controller.go:175]: WDYT about including the cluster in this event, like we include the addon name in the event below?

File: api/pkg/crd/kubermatic/v1/cluster.go (hunk 1 above)
Comment (xrstf) [reply api/pkg/crd/kubermatic/v1/cluster.go:112]: Without. The comments add nothing and auto-generated comments are pointless.

File: api/pkg/controller/addon/addon_controller.go (hunk 2 above)
Comment (alvaroaleman) [reply api/pkg/controller/addon/addon_controller.go:175]: Addons are namespaced and created per cluster, so that is implicit

**Pull request merged**